include/emp/prefab/ControlPanel.hpp

```diff
@@ -0,0 +1,331 @@
+    : ControlPanel(
+      refresh_rate,
+      refresh_unit,
+      new internal::ControlPanelInfo(in_id)
+    ) { ; }
+
+    /**
+     * Set the simulation for this control panel
+     * @param sim the function to be run every frame (as fast as possible)
+     */
+    ControlPanel & SetSimulation(const std::function<void()> & sim) {
+      Info()->SetSimulation(sim);
+      return *this;
+    }
+
+    /**
+     * Get the simulation for this control panel
+     * @return the function to be run every frame (as fast as possible)
+     */
+    const std::function<void()> & GetSimulation() const {
+      return Info()->GetSimulation();
+    }
+
+    /**
+     * Set the refresh rate units for this control panel.
+     * @param unit either "MILLISECONDS" or "FRAMES"
+     * @note rates are independent for "MILLISECONDS" and "FRAMES"
+     */
+    ControlPanel & SetRefreshUnit(const std::string & units) {
+      Info()->SetUnit(units);
+      return *this;
+    }
+
+    /**
+     * Set the refresh rate for this control panel for the current unit.
+     * @param rate period in frames or milliseconds
+     * @note rates are independent for "MILLISECONDS" and "FRAMES".
+     */
+    void SetRefreshRate(const int & rate) {
+      Info()->SetRate(rate);
+    }
+
+    /**
+     * Set the refresh rate for this control panel.
+     * @param rate the number of milliseconds or frames between refreshes
+     * @param unit either "MILLISECONDS" or "FRAMES"
+     * @note rates are independent for "MILLISECONDS" and "FRAMES"
+     */
+    void SetRefreshRate( const int & rate, const std::string & units) {
+      Info()->SetUnit(units);
+      Info()->SetRate(rate);
+    }
+
+    /**
+     * Adds a Widget to a list of widgets redrawn at the specified refresh rate.
+     * @param area a widget
+     */
+    void AddToRefreshList(Widget & area) {
+      Info()->GetRefreshList().push_back(area);
+    }
+
+    /**
+     * Stream operator to add a component to the control panel.
+     *
+     * Some special behavior: Buttons and ToggleButtonGroups will be added
+     * to the last ButtonGroup added to keep related components together.
+     * If you want to start a new group, just stream in a new ButtonGroup.
+     * @param in_val a component to be added to the control panel
+     */
+    template <typename IN_TYPE>
+    ControlPanel & operator<<(IN_TYPE && in_val) {
+      // Took soooo long to figure out but if in_val is a r-value ref
+      // IN_TYPE is just the TYPE. If it's l-value then it's TYPE &.
+      // std::decay and forward help handle both.
+      if constexpr(std::is_same<typename std::decay<IN_TYPE>::type, web::Button>::value ||
+        std::is_same<typename std::decay<IN_TYPE>::type, ToggleButtonGroup>::value) {
+        button_line << std::forward<IN_TYPE>(in_val);
+      } else if constexpr(std::is_same<typename std::decay<IN_TYPE>::type, ButtonGroup>::value) {
+        button_line = std::forward<ButtonGroup>(in_val);
```

Comment:
std::forward<IN_TYPE> ? (unless you have a good reason to deviate from that idiom)